JavaBytecodeCompiler/src/main/java/org/openzen/zenscript/javabytecode/compiler/JavaMethodBytecodeCompiler.java

```diff
@@ -1390,20 +1390,28 @@ public Void specialStaticMethod(JavaSpecialMethod method, CallArguments args) {
 				Expression value = arguments[0];
 				value.accept(expressionVisitor);
 
+
+
+				// for .toArray(new T[0]) we need to know what T is
+				final TypeID arrayBaseType = value.type
```

Comment:
While this does work for the Java collection types, there's probably a better way to go about this?